config/helm/kustomization.yaml

```diff
@@ -1,100 +1,116 @@
+    application.giantswarm.io/branch: "{{ .Values.project.branch }}"
+    application.giantswarm.io/commit: "{{ .Values.project.commit }}"
+    application.giantswarm.io/team: "{{ index .Chart.Annotations \"application.giantswarm.io/team\" }}"
 
 images:
-  - name: registry.k8s.io/cluster-api-azure/cluster-api-azure-controller
-    newName: "{{.Values.image.registry}}/{{.Values.image.name}}"
-    newTag: "{{.Values.image.tag}}"
-  - name: mcr.microsoft.com/k8s/azureserviceoperator:v2.4.0
-    newName: "{{.Values.aso.image.registry}}/{{.Values.aso.image.name}}"
+# Cluster API Provider Azure
+- name: registry.k8s.io/cluster-api-azure/cluster-api-azure-controller
+  newName: "{{ .Values.image.registry }}/{{ .Values.image.name }}"
+  newTag: "{{ .Values.image.tag }}"
+# Azure Service Operator
+- name: mcr.microsoft.com/k8s/azureserviceoperator
+  newName: "{{ .Values.aso.image.registry }}/{{ .Values.aso.image.name }}"
 
-patchesStrategicMerge:
-  - deployment-capz-args.yaml
-  - deployment-affinity.yaml
-  - deployment-environment.yaml
-  - deployment-metrics-port.yaml
-  - deployment-securitycontext.yaml
-  - deployment-toleration.yaml
-  - deployment-aso-securitycontext.yaml
-  - deployment-aso-volume.yaml
-  - webhook-certificate.yaml
-  - webhook-certificate-aso.yaml
-  - service-add-metrics-port.yaml
-  - certificate.yaml
-  - certificate-serviceoperator.yaml
-  - delete-selfsigned-cert-issuer.yaml
-  - delete-ns.yaml
-  - delete-nmi.yaml
-  - secret-aso-controller-settings.yaml
+transformers:
+# Mutating webhooks
+- patches/mutatingwebhooks/zzz-prefix.yaml
 
 patches:
-- path: deployment-aso-remove-kube-proxy.yaml
-- path: delete_aadpodidentity_crds.yaml
+# CRDs
+- path: patches/crds/all.yaml
   target:
+    group: apiextensions.k8s.io
+    version: v1
     kind: CustomResourceDefinition
-    annotationSelector: api-approved.kubernetes.io=unapproved
-- path: crd_cainjection.yaml
-  target:
-    kind: CustomResourceDefinition
-    labelSelector: cluster.x-k8s.io/provider=infrastructure-azure
-- path: aso_crd_cainjection.yaml
+- path: patches/crds/azureserviceoperator.yaml
   target:
+    group: apiextensions.k8s.io
+    version: v1
     kind: CustomResourceDefinition
     labelSelector: app.kubernetes.io/name=azure-service-operator
-- path: crd_webhook.yaml
+- path: patches/crds/capz.yaml
   target:
+    group: apiextensions.k8s.io
+    version: v1
     kind: CustomResourceDefinition
     labelSelector: cluster.x-k8s.io/provider=infrastructure-azure
-- path: crd_clusterctl_labels.yaml
+- path: patches/crds/unapproved.yaml
   target:
+    group: apiextensions.k8s.io
+    version: v1
     kind: CustomResourceDefinition
-    labelSelector: cluster.x-k8s.io/provider=infrastructure-azure
-- target:
-    kind: ValidatingWebhookConfiguration
-  path: validating-webhook-watchfilter.yaml
-- target:
-    kind: MutatingWebhookConfiguration
-  path: mutating-webhook-watchfilter.yaml
-- target:
+    annotationSelector: api-approved.kubernetes.io=unapproved
```

Comment:
What is "unapproved" in this context? 

Does this cause all CRDs with that annotation to be deleted? I've not seen this kinda thing before.

Reply:
First of all: Yes, it deletes all the CRDs with this annotation, so they do not show up in the resulting Helm chart and therefore do not get installed. I _think_ unapproved CRDs means created and used by the developers of the CAPZ controllers, but not yet approved by the Kubernetes API SIG, so probably some pre-alpha stuff. I can imagine us not to install and provide them to customers as soon as they are not approved and therefore still subject to possible change and resulting friction.

Again: I tried not to change the resulting Helm chart, only the way it's being generated.